### PR TITLE
Add release notes for 0.229

### PR DESCRIPTION
**Pull request by caithagoras**
# Missing Release Notes
## Ajay George
- [x] https://github.com/jzmq/presto/pull/41 [pull] master from prestodb:master
- [x] https://github.com/jzmq/presto/pull/18 [pull] master from prestodb:master

## James A. Gill
- [x] c006c8075038aa5ba73f4f7b388fa8b2ab7da245 Fix GeometryToBingTiles for certain geometries

## Ke Wang
- [x] b4752ccb4ee2a1717df903d1a0fed110a2df6b18 fix organizationDiscoveryIntervalMillis bug

## Leiqing Cai
- [x] 7e12d427f748776c47a38eec8f406d528e208246 Add JSON annotations to SqlFunctionId

## Rongrong Zhong
- [x] https://github.com/prestodb/presto/pull/13384 Function implementation (Merged by: Rongrong Zhong)
- [x] https://github.com/prestodb/presto/pull/13615 Revert HashBuilderOperator changes (Merged by: Rongrong Zhong)

## Saksham Sachdev
- [x] https://github.com/prestodb/presto/pull/13526 Implement Basic Example For Jdbc Row Expression Translation (Merged by: James Sun)

## Yi He
- [x] https://github.com/prestodb/presto/pull/13112 Row expression predicate pushdown (Merged by: James Sun)
- [x] https://github.com/prestodb/presto/pull/13560 Add RowExpressionPredicateExtractor (Merged by: James Sun)

## oerling@fb.com
- [x] https://github.com/prestodb/presto/pull/13631 Fix position count of prefilled Blocks (Merged by: Maria Basmanova)
- [x] https://github.com/prestodb/presto/pull/13630 Fix getBlockView for all-null SliceDictionarySelectiveReader (Merged by: Maria Basmanova)
- [x] https://github.com/prestodb/presto/pull/13607 Use fast varint read path in LongInputStreamV1 (Merged by: James Sun)
- [x] https://github.com/prestodb/presto/pull/13505 Intersect pushdown predicates (Merged by: Rebecca Schlussel)

# Extracted Release Notes
- #13379 (Author: Vic Zhang): Add peak memory distribution among tasks of each stage
  - Add peak total memory distribution among tasks of each stage.
- #13477 (Author: Cem Cayiroglu): Support max_tasks_per_stage for scan
  - Respect stage.max-tasks-per-stage to limit number of tasks for scan.
- #13504 (Author: Devesh Agrawal): Adding presto-pinot with pushdown working
  - Added a new Pinot connector.
  - Split `ConnectorPlanOptimizerProvider` to have two phases for connectors to participate query optimization: `LOGICAL` and `PHYSICAL`. The two phases correspond to pre- and post-shuffle optimization respectively.
- #13527 (Author: Swapnil Tailor): Passing X_Forwarded_For in header from proxy server
  - Start forwarding X_Forwarded_For in header from Proxy.
- #13535 (Author: Jiexi Lin): Support using remote HDFS as storage in Raptor
  - Change `storage.data-directory` from path to URI. For existing deployment on local flash, a scheme header "file://" should be added to the original config value.
  - Change error code name `RAPTOR_LOCAL_FILE_SYSTEM_ERROR` to `RAPTOR_FILE_SYSTEM_ERROR`.
- #13546 (Author: James Sun): Move SetOperationNode to SPI
  - Move `SetOperationNode` to SPI. Connectors can now alter query plan to have union, intersect, and except.
- #13561 (Author: Leiqing Cai): Support CREATE FUNCTION execution
  - Add support for ``CREATE FUNCTION``.
- #13584 (Author: Nikhil Collooru): Fail query with large taskUpdate size
  - Add configuration property ``experimental.internal-communication.max-task-update-size`` to limit the size of the taskUpdate.
- #13592 (Author: Yi He): Add common InternalCommunicationConfig to turn on HTTPS
  - Added configs to use common HTTPS settings for internal communications.
- #13594 (Author: James Petty): Parquet Dictionary Predicate Pushdown Fixes
  - Fix parquet predicate pushdown on dictionaries to consider more than just the first predicate column.
  - Improve parquet predicate pushdown on dictionaries to avoid reading additional data after successfully eliminating a block.
- #13610 (Author: Wenlei Xie): Add non-linear transformation for structural hash
  - Improve hash value computation for structrual types to avoid collision.
- #13612 (Author: James Gill): Add geospatial function line_interpolate_point
  - Add geospatial function line_interpolate_point.

# All Commits
- b4752ccb4ee2a1717df903d1a0fed110a2df6b18 fix organizationDiscoveryIntervalMillis bug (Ke Wang)
- c006c8075038aa5ba73f4f7b388fa8b2ab7da245 Fix GeometryToBingTiles for certain geometries (James A. Gill)
- 7e12d427f748776c47a38eec8f406d528e208246 Add JSON annotations to SqlFunctionId (Leiqing Cai)
- fbdbeeb03c508f628ef294a05538aa3f583e24f6 Add geospatial function line_interpolate_point (James Gill)
- f424e1b17c6c617f5523c9c739d23101bdf695e7 Fix slice reader reading empty strings column (Bhavani Hari)
- 5d8105c835aedeca5791a9ece27057e5b9d2049f Fix slice reader check for presentStream (Bhavani Hari)
- b82b56b3e868742b9a72784f61f512cb36c888d4 Remove TypeProvider from FilterStatsCalculator.filterStats (Sahar Massachi)
- 34c846e37a5545f359e868dd19d658096877ec40 Ensure TableFinishOperator input is ordered (Andrii Rosa)
- 62bf6986d1955e164de11a627f79c39859bb83cf Remove synchronized method in FixedSourcePartitionedScheduler (Shixuan Fan)
- c61d6bb048881d44dbbd9f9aefb6e5bcb4e010e1 Make updateTaskStatus not synchronized within SqlStageExecution (Andrii Rosa)
- be1db83343333f65c4c8612991fa964ec712354d Fix recoverable grouped execution for task_writer_count=1 (Andrii Rosa)
- dd00be96c20b31d09130b1186d106a52b7536225 Improve TestHiveRecoverableGroupedExecution (Andrii Rosa)
- 91fb5c1aa82b7ad30f0ebba34f209580ac216bda Fix NPE in SimpleNodeSelector (Andrii Rosa)
- 9fd0400fcb9e305b426e717198b1d870f29e90c0 Add non-linear transformation for structural hash (Wenlei Xie)
- 241392469199c058aab4f6bdf3944a3cfa2e4068 Fix position count of prefilled Blocks (oerling@fb.com)
- 1373a9ff23ef7e1c96ea901682f7d3fc7ba41a7f Fix getBlockView for all-null SliceDictionarySelectiveReader (oerling@fb.com)
- 808043fe0b009c9c1122fa20766ddb104ae71fc6 Add cache for stripe metadata (Shixuan Fan)
- 177a680faafebd3d969e1647701ed55c22023e80 Move cache related class into separate package (Shixuan Fan)
- e1f0dd6d420a514b0e2d5db672b1e3958b7e2908 Remove unused method in OrcTestingUtil (Shixuan Fan)
- f24a4524b9cbe9a1b29e341ba2110ec707ba5caf Fix an issue when fetching function implementation in FunctionManager (Leiqing Cai)
- dabecccdda52aa56224dacd7f7d63a2d76c0eba6 Support CREATE FUNCTION execution (Leiqing Cai)
- 866f66577903dbbb55a412e32857ff46aa03a186 Add support for refreshing functions cache in namespace manager (Leiqing Cai)
- 109ad5e3ce322e7ef19a5cb683882c60503e032f Rename FunctionNamespaceManager#rollback to #abort (Leiqing Cai)
- 3b7f14306a6f82b603d335ac90be45a1b5dabb2d Add ParametersAreNonnullByDefault annotation to cache loading methods (Leiqing Cai)
- 02b8c0e2abe15825409e197e01b638080d1d5da3 Move SqlInvokedRegularFunctionTestUtils to presto-sql-functions (Leiqing Cai)
- e81b812b91b019e589ca1520eef22cdaf3735fbc Move classes related to SQL-invoked functions into spi (Leiqing Cai)
- a13b4c57bd828d91764771b412c499a6223d3dbb Rename SqlInvokedRegularFunction#comment to description (Leiqing Cai)
- dd7f571d2d27c049384c14d72fdf3cbdef346c0d Require function namespace to be exactly 2 parts (Leiqing Cai)
- 4051fd7ac165d841ac03ff01adca954862589968 Move spi.relation.FullyQualifiedName to spi.QualifiedFunctionName (Leiqing Cai)
- ab9d0ec22584ce1ab134ef2580514ab4791b6f25 Remove unused FeatureConfig from PlanOptimizers (James Sun)
- 694bd5dc61c77bf10fe83c6df3438bf92996997a Support SQL function execution (Rongrong Zhong)
- d47c75d0203013eafbc3b54afea88c17a23d60ed Add ScalarFunctionImplementation abstraction (Rongrong Zhong)
- 7294fa5018239f6f145ad9eef30634ac5884e77c Add funcion implementation type and argument names to FunctionMetadata (Rongrong Zhong)
- 04ead21d85198984cb3fdedb078603fa0f62196c Rename ScalarFunctionImplementation to BuiltInScalarFunctionImplementation (Rongrong Zhong)
- b0645420b27f47dbc7feb67186621a9cc712cba5 Add SqlFunctionProperties to bytecode compiler cache keys (Rongrong Zhong)
- a5701caa56da7447be42dc4e388cb649e799a9e5 Remove Session from SqlToRowExpressionTranslator (Rongrong Zhong)
- a80dea946b982eecf7a4eb718fdc5d18094aa4f7 Factor out Expression related session configs into SqlFunctionProperties (Rongrong Zhong)
- 297e2bac161281adc1e9b4601fbd8a6e188b38c0 Fix error message in FunctionManager (Rongrong Zhong)
- c1a0c65d85987b13beb20f1b09343760beb5f3f9 Check call convention once in RowExpressionInterpreter.visitCall (Rongrong Zhong)
- 096c1def104088264d2d3ce3432c5b8ab64ab678 Check call convention once in ExpressionInterpreter.visitFunctionCall (Rongrong Zhong)
- 929e8a90e855952175101b350951c479ae8c6b48 Remove TypeProvider from LocalExecutionPlanContext (Rongrong Zhong)
- b54d2483c76afe2d7b57c7ae56c356ba25440e88 Improve HashGenerator.getPartition performance (Ying Su)
- 898037b78649c2c16314eb154ea1b765be3b7a1e Fix hive columnIndex with default partition (Bhavani Hari)
- 052e0eb6b79e235d234fb3cbed6a07fdd57d8ab6 Use URI in raptor.properties (Jiexi Lin)
- 2be543e0e1a993b8c4619247c14b25c43cf1a6df Optimize PlanFragment serialization (Nikhil Collooru)
- af3b9dd74de1c31cbd71fa7a78ddb673185ff263 Revert " Include predicted size of built hash in HashBuilderOperator" (Rongrong Zhong)
- 2404248fca3f28ff8d7cd0a7c52b6a16bf11d56e Revert "Do not overcommit memory in HashBuilderOperator" (Rongrong Zhong)
- b0fed6e50b842deedf5ca9d63ff0909ce16de7a5 Make Polygon WKTs valid in test cases (James A. Gill)
- d28e5de3f932510dae5c52568b0ce7fcddc895fd Improve Geometry Serialization Tests (James Gill)
- de1dda7a12921d65fdeb320ef266bd20f4efb24a Improve error message when transaction is gone (Andrii Rosa)
- a140fba74e14c612ecaaa83b1ee78be2c6c3c760 Refactor Hive transaction management not to use metastore getter (Andrii Rosa)
- e6b768774317690ec14d82cc5b2dd8de8b9f4444 Do not update query failureCause when query is FINISHING (Andrii Rosa)
- 2b5711cb7540a84411120124a25f6ac3ace88214 Do not fail finishing query unless the transaction failed (Andrii Rosa)
- aa47ca31869fcc7b1480772d5bf0af4e536fe913 Check if transaction still exist on transitioning to failed query (Andrii Rosa)
- d75bd6b888399af7e055fad207be4b38289ef1e9 Use fast varint read path in LongInputStreamV1 (oerling@fb.com)
- 5a143aa42845b4c34af6ac8d617f3989ca3e041f Export round trip count for update/info/status (Wenlei Xie)
- 78f24709b80f17bc7b7f6a0317cd9ee0515c638e Adding presto-pinot with pushdown (Devesh Agrawal)
- 6a7c4c81957cb4a85bc316bacf597c9fbe568f34 Add common InternalCommunicationConfig to turn on HTTPS (Yi He)
- 0be3619feda6f86f3bcad7fc7cabab49be56494b Fix table scan results with disjoint pushdown filters (Bhavani Hari)
- 2b79a586d18477faa64f4bb6addabf2f8e8d68f5 Refactor SqlStageExecution creation (Rebecca Schlussel)
- ad92616433964f38f1d463a8091b56ece94f14db Small cleanups of SqlQueryScheduler (Rebecca Schlussel)
- 314dd12c9d9efbd36b478b25636a3b832a2a0428 Add FileOpener interface in Hive connector (Shixuan Fan)
- 6098d199bd1c8f0e03cb342d91d88525048c77f8 Remove unused static variable in TestBackgroundHiveSplitLoader (Shixuan Fan)
- 2de30ec442f1ef9ded4e658ddbbc300ecaaf62c9 Fail query with large taskUpdate size (Nikhil Collooru)
- 88416e540a15e12bab7325474dca68f0cd20b794 Add peak memory distribution among tasks of each stage (Vic Zhang)
- a97dd9406206a92754998072f4e908f6615b32f0 Rename StageCpuDistribution to ResourceDistribution (Vic Zhang)
- 6dbfefcb7aa82e7864ab751f7bed72de8bd9e542 Add peak total memory to TaskStats (Vic Zhang)
- 947b5b90bbf8c4d66dcff5432cd9625c115e6918 Add PartitionObjectBuilder interface for Hive (Ajay George)
- 93506020afa3e290618f438e01a2554e3e6bb0e4 Support max_tasks_per_stage for scan (Cem Cayiroglu)
- a71977ea0a3b0eb6488578e839afedb64bbcde58 Stop Parquet Dictionary Reads Eagerly in Predicate Pushdown (James Petty)
- d257b0c46a7e377bae7a919d607070a0fbbbaca0 Fix Parquet Predicate Pushdown on Dictionaries (James Petty)
- fcd6f04d84887d9b51013353567f509d3efcf01e Do not overcommit memory in HashBuilderOperator (Andrii Rosa)
- a38dbfa332f4aee567d05814f20574cf7ac4999d Remove DeleteHandle from DeleteNode (Rebecca Schlussel)
- b581ffb91ac8a7707ba5f85004745d5801e1b0c2 Use TableHandle directly in StatisticsWriterNode (Rebecca Schlussel)
- 928a23437a4e4cdece2bf3e6cebf7814767ba7fe Move BeginTableWrite logic to scheduler (Rebecca Schlussel)
- 7ebe52287206aef8ddf91d60630d2c9cedf95f99 Simplify code dealing with WriterTargets (Rebecca Schlussel)
- 5b8ec08aadf8226616c3e922deda5e81375c6522 Add ConnectorId to CreateName (Rebecca Schlussel)
- 94c73cc0be438a1437de59997b309a41b6b55971 Make WriterTargets optional in plan nodes (Rebecca Schlussel)
- efbd394cc3b910b9aa5b60f1b6954823f0cff726 Replace DeleteHandle with TableHandle for metadata delete (Rebecca Schlussel)
- 48c06932c9c032a2ac3b557b9babef6e56c232a7 Remove unused field from LocalExecutionPlanner (Rebecca Schlussel)
- a9fa7b6c266fe30c2f42ba575b01542d9258b08c Remove test usage of TableScan json constructor (Rebecca Schlussel)
- 44bc89782cc7bdc8b02b037608bbf63d5628f700 Fix Flaky TestTransactionManager.testExpiration (Cem Cayiroglu)
- fbb7794517b124e373af863a423aad9f68a259b3 Support using HDFS as storage in Raptor (Jiexi Lin)
- 9fb46ce831b4a20fa7e91e29ab6701a36e4ed037 Revert reserved words introduced in CREATE FUNCTION syntax support (Leiqing Cai)
- 0e7847b69106120b7d437707ac9412da2b0778d5 Fix SliceReader ArrayOutOfBounds when presentStream is null (Bhavani Hari)
- a6fd61b38336d57dc85ffa2cd336cbed13c1712a Move RowExpression translation above AddExchange (James Sun)
- 55c3f7c43437a247f122053fe5b54b7bb20a1a76 Rename SymbolResolver to VariableResolver (James Sun)
- 332d48418da295b0da4de3e34f9c164486cabb73 Move SetOperationNode to SPI (James Sun)
- 7491f2038cc55a0feba5c6d44e44a5f2b2ae4bfd Add session property for nested subfield filters (Bhavani Hari)
- f62dfdd47f83105e7695275bbf2b6d5f21b34b17 Hanlde isNull filter for Array type (Bhavani Hari)
- 9092cb7b41638de8c5ec4046b632be89f66c5e7b Add example for Jdbc Connector Row Expression Translation (Saksham Sachdev)
- e3694a0dd43403d2f5ae56107582c937f7408be7 Fix compareRanges logic (Bhavani Hari)
- 49df615aaffddab28f47144951c83c003dedab2f Move TranslateExpressions below AddExchange (Yi He)
- 4f74189285f63bb6146a9e0bd3ab9b5a3ec362d0 Move TranslateExpressions above last predicatePushdown (Yi He)
- 434357405e40c4e573178d3d20e28cd0c57f98cf Include error message in SimpleHttpResonseHandler for SMILE (Nikhil Collooru)
- a51c1e3cabb20b162e8c108611daf5327be68073 Generate unique index for PartitionColumns (Bhavani Hari)
- 90cfbd2345f546087fce57932852e05efcd3895f Passing X_Forwarded_For from proxy (Swapnil Tailor)
- 8d5d5e67e1e2276e9e2a1fc02f471e6d0a020c89 Update to Facebook airlift (Wenlei Xie)
- 184fb504592d8ba432d6904ae0dd36ec34aa9457 Add ignored dependencies in presto-hive (Wenlei Xie)
- cff2383bb95827c61a65e39e1d359dcd3e6ee93d Add more dependencies for HiveGcsModule (Wenlei Xie)
- 442a46f6c5c0a69bc415259fffaf3624b0ab41e6 Fix undeclared dependency used for HTTP constants (David Phillips)
- 42e7b22d5e4362072858f2f5c30fad913effa686 Use ValidationAssertions in TestPowerOfTwoValidator (David Phillips)
- 7e2f60dc9ab7ae9b68dccfa20d0c5ae4b34ab7cd Reset DictionaryBlock values after getBlock (Bhavani Hari)
- 270a40f087435d6ca7548d0f2d73a79021e2dd08 Fix selective readers output positions when reading null streams (Bhavani Hari)
- c1b2b2e480afbcdc5215cb2d19352840f770da15 Add cache for orc file tail information (Shixuan Fan)
- 418d63b634f9af16e4d8831f2f86c6018cb02a3e Add RowExpressionPredicateExtractor (Yi He)
- 9a6d4de9a7b0389ca1b7807e164ff5759e69a983 Intersect pushdown predicates (oerling@fb.com)
- 3c8584ff76c6d435773790a3b31597d689f90aed prepare for next development iteration - 0.229-SNAPSHOT (Rongrong Zhong)

### CODE REVIEW

File: presto-docs/src/main/sphinx/release.rst [hunk 1]
```diff
@@ -6,6 +6,7 @@ Release Notes
     :maxdepth: 1
 
     release/release-0.228
+    release/release-0.229
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/release.rst]: This should be before 0.228?

File: presto-docs/src/main/sphinx/release/release-0.229.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
+=============
+Release 0.229
+=============
+
+General Changes
+_______________
+* Add geospatial function line_interpolate_point.
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/release/release-0.229.rst]: `:func:line_interpolate_point`

File: presto-docs/src/main/sphinx/release/release-0.229.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+* Improve hash value computation for structrual types to avoid collision.
+* Add support for ``CREATE FUNCTION``.
+* Added configs to use common HTTPS settings for internal communications.
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/release/release-0.229.rst]: `Add`. Do we need to be more specific on this?

File: presto-docs/src/main/sphinx/release/release-0.229.rst [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+* Add configuration property ``experimental.internal-communication.max-task-update-size`` to limit the size of the taskUpdate.
+* Add peak total memory distribution among tasks of each stage.
+* Respect stage.max-tasks-per-stage to limit number of tasks for scan.
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/release/release-0.229.rst]: ` ``stage.max-tasks-per-stage`` `

File: presto-docs/src/main/sphinx/release/release-0.229.rst [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+* Move `SetOperationNode` to SPI. Connectors can now alter query plan to have union, intersect, and except.
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/release/release-0.229.rst]: ``

File: presto-docs/src/main/sphinx/release/release-0.229.rst [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+* Start forwarding X_Forwarded_For in header from Proxy.
+
+SPI Changes
+___________
+* Added a new Pinot connector.
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/release/release-0.229.rst]: This is a connector change, not an SPI change.

File: presto-docs/src/main/sphinx/release/release-0.229.rst [hunk 7]
```diff
@@ -0,0 +1,30 @@
+=============
+Release 0.229
+=============
+
+General Changes
+_______________
+* Add geospatial function line_interpolate_point.
+* Improve hash value computation for structrual types to avoid collision.
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/release/release-0.229.rst]: Optimizations should be before additions (fix > optimization > addition > deletion): https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines#order-of-changes

**Comment by jagill** [comment]
For commit  c006c80 (it didn't seem to auto-extract :( )

General Changes
* Fix #13639: geometry_to_bing_tile fails on certain degenerate geometries.

**Comment by sachdevs** [comment]
For #13526: There is no release note needed, it does not add any new functionality at all. Will include NO RELEASE NOTE next time.

**Comment by caithagoras** [comment]
@rongrong I should probably make the auto-generated PR title to have `[WIP]` because it is not yet ready for review. I was waiting for the missing release notes before working on formatting it.

**Comment by caithagoras** [comment]
@sachdevs The correct format will be to include the following in your PR description. Thanks!
```
== RELEASE NOTES ==
```

File: presto-docs/src/main/sphinx/release/release-0.229.rst [hunk 8]
```diff
@@ -0,0 +1,34 @@
+=============
+Release 0.229
+=============
+
+General Changes
+_______________
+* Fix an issue that would cause query failure when calling :func:`geometry_to_bing_tile` on certain degenerate geometries.
```
Comment (caithagoras) [inline presto-docs/src/main/sphinx/release/release-0.229.rst]: @jagill Is this updated release note accurate? Does the "fail" you mentioned means "query failure"?

File: presto-docs/src/main/sphinx/release/release-0.229.rst [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+* Add geospatial function :func:`line_interpolate_point`.
+* Add support for ``CREATE FUNCTION``.
+* Add support for using configuration property ``stage.max-tasks-per-stage`` to limit the number of tasks for scan.
+* Add support for passing ``X_Forwarded_For`` header from Proxy to coordinator.
+* Add configuration to use common ``HTTPS`` settings for internal communications.
```
Comment (caithagoras) [inline presto-docs/src/main/sphinx/release/release-0.229.rst]: @hellium01 What are the specific configuration properties that we added?

File: presto-docs/src/main/sphinx/release/release-0.229.rst [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+* Add configuration property ``experimental.internal-communication.max-task-update-size`` to limit the size of the ``TaskUpdate``.
+* Add peak total memory distribution among tasks of each stage.
```
Comment (caithagoras) [inline presto-docs/src/main/sphinx/release/release-0.229.rst]: @viczhang861 It's unclear to me what this means. Where are we adding the peak total memory distribution? Are we adding it to the query stats?

Comment (caithagoras) [reply presto-docs/src/main/sphinx/release/release-0.229.rst]: Also, what does "among tasks" means? Are we adding the distribution for each task or for all tasks?

File: presto-docs/src/main/sphinx/release/release-0.229.rst (hunk 9 above)
Comment (hellium01) [reply presto-docs/src/main/sphinx/release/release-0.229.rst]: Following configuration options are added:
```
internal-communication.https.trust-store-path
internal-communication.https.included-cipher
internal-communication.https.excluded-cipher
```

**Comment by rongrong** [comment]
> @rongrong I should probably make the auto-generated PR title to have `[WIP]` because it is not yet ready for review. I was waiting for the missing release notes before working on formatting it.

That's fine. Just want to speed up the process since we are on a tight schedule. The review is free anyways. 😛

File: presto-docs/src/main/sphinx/release/release-0.229.rst (hunk 10 above)
Comment (viczhang861) [reply presto-docs/src/main/sphinx/release/release-0.229.rst]: Let's change it to "Add peak task memory distribution for each stage to QueryStatistics."

**Comment by caithagoras** [comment]
@rongrong Ready for final review.

File: presto-docs/src/main/sphinx/release/release-0.229.rst [hunk 11]
```diff
@@ -0,0 +1,36 @@
+=============
+Release 0.229
+=============
+
+General Changes
+_______________
+* Fix an issue that would cause query failure when calling :func:`geometry_to_bing_tiles` on certain degenerate geometries.
+* Add geospatial function :func:`line_interpolate_point`.
+* Add support for ``CREATE FUNCTION``.
+* Add support for using configuration property ``stage.max-tasks-per-stage`` to limit the number of tasks for scan.
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/release/release-0.229.rst]: `Add configuration property ...`

File: presto-docs/src/main/sphinx/release/release-0.229.rst [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+* Add support for passing ``X_Forwarded_For`` header from Proxy to coordinator.
+* Add configuration property ``experimental.internal-communication.max-task-update-size`` to limit the size of the ``TaskUpdate``.
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/release/release-0.229.rst]: group all configuration property changes together?

File: presto-docs/src/main/sphinx/release/release-0.229.rst [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+* Add configuration properties ``internal-communication.https.trust-store-path``, ``internal-communication.https.included-cipher``, and ``internal-communication.https.excluded-cipher``
+  to easily set common https configurations for all internal communications at one place.
+* Add peak task memory distribution of each stage to ``QueryStatistics``.
+
+Pinot Connector Changes
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/release/release-0.229.rst]: Put this section together with the other connector changes after SPI changes?

File: presto-docs/src/main/sphinx/release/release-0.229.rst [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+_______________________
+* Add Pinot connector.
+
+SPI Changes
+___________
+* Add support for connectors to alter query plans involving ``UNION``, ``INTERSECT``, and ``EXCEPT``, by moving ``SetOperationNode`` to SPI.
+* Improve interface ``ConnectorPlanOptimizerProvider`` to allow connectors to participate in query optimization in two phases, ``LOGICAL`` and ``PHYSICAL``.
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/release/release-0.229.rst]: Is this line a little too long? Maybe some others as well.

File: presto-docs/src/main/sphinx/release/release-0.229.rst (hunk 11 above)
Comment (caithagoras) [reply presto-docs/src/main/sphinx/release/release-0.229.rst]: The configuration property already exists and is being used. We're respecting its value in a different circumstance.

File: presto-docs/src/main/sphinx/release/release-0.229.rst (hunk 14 above)
Comment (caithagoras) [reply presto-docs/src/main/sphinx/release/release-0.229.rst]: I generally don't find this is an issue in the release notes as long as they're readable on Github and in IntelliJ. User will see generated HTML, which is irrelevant the line length here.

File: presto-docs/src/main/sphinx/release/release-0.229.rst [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+Release 0.229
+=============
+
+General Changes
+_______________
+* Fix an issue that would cause query failure when calling :func:`geometry_to_bing_tiles` on certain degenerate geometries.
+* Add geospatial function :func:`line_interpolate_point`.
+* Add support for ``CREATE FUNCTION``
+* Add support for passing ``X_Forwarded_For`` header from Proxy to coordinator.
+* Add support for using configuration property ``stage.max-tasks-per-stage`` to limit the number of tasks for scan.
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/release/release-0.229.rst]: `Add configuration property ...`

File: presto-docs/src/main/sphinx/release/release-0.229.rst (hunk 11 above)
Comment (rongrong) [reply presto-docs/src/main/sphinx/release/release-0.229.rst]: Maybe `Add support to respect xxx` then?

Comment (caithagoras) [reply presto-docs/src/main/sphinx/release/release-0.229.rst]: Changed to 
```
Add support to respect configuration property ``stage.max-tasks-per-stage`` for limiting the number of tasks per scan.
```

**Comment by caithagoras** [comment]
The CLA check seems to have passed.

**Pull request merged**